mondrian/src/main/java/mondrian/xmla/XmlaHandler.java

```diff
@@ -797,6 +799,43 @@ private void execute( XmlaRequest request, XmlaResponse response ) throws XmlaEx
 
         if ( queryPart instanceof DrillThrough ) {
           result = executeDrillThroughQuery( request );
+        } else if ( queryPart instanceof DmvQuery ) {
+          DmvQuery dmvQuery = (DmvQuery) queryPart;
+
+          HashMap<String, String> upperCaseProperties = new HashMap<>();
+          for ( String key : request.getProperties().keySet() ) {
+            String newKey = null;
+            if ( key != null ) {
+              newKey = key.toUpperCase();
+            }
+            upperCaseProperties.put( newKey, request.getProperties().get( key ) );
+          }
+
+          HashMap<String, Object> restrictions = new HashMap<>();
+          if ( upperCaseProperties.containsKey( PropertyDefinition.Catalog.name().toUpperCase() ) ) {
+            List<String> restriction = new ArrayList<>();
+            restriction.add( request.getProperties().get( "Catalog" ) );
+            restrictions.put( Property.StandardMemberProperty.CATALOG_NAME.name(), restriction );
+          }
+          mondrian.xmla.impl.DmvXmlaRequest dmvXmlaRequest = new mondrian.xmla.impl.DmvXmlaRequest(
```

Comment:
Does this need the full package here or could it be simpler like only `DmvXmlaRequest`?